instance/models/mixins/database.py

```diff
@@ -69,28 +108,55 @@ def provision_mysql(self):
+
+            # Create default databases and users, and grant privileges
+            for database in self.mysql_databases:
                 # We can't use the database name in a parameterized query, the
-                # driver doesn't escape it properly. Se we escape it here instead
-                database_name = connection.escape_string(database).decode()
-                cursor.execute('CREATE DATABASE `{0}` DEFAULT CHARACTER SET utf8'.format(database_name))
-                cursor.execute('CREATE USER %s IDENTIFIED BY %s', (self.mysql_user, self.mysql_pass,))
-                cursor.execute('GRANT ALL ON `{0}`.* TO %s'.format(database_name), (self.mysql_user,))
+                # driver doesn't escape it properly. So we escape it here instead
+                database_name = connection.escape_string(database["name"]).decode()
+                _create_database(cursor, database_name)
+                user = database["user"]
+                _create_user(cursor, user, self._get_mysql_pass(user))
+                privileges = database.get("priv", "ALL")
+                _grant_privileges(cursor, database_name, user, privileges)
+                _grant_privileges(cursor, database_name, self.migrate_user, "ALL")
+                additional_users = database.get("additional_users", [])
+                for additional_user in additional_users:
+                    _grant_privileges(cursor, database_name, additional_user["name"], additional_user["priv"])
+
+            # Create read_only user with appropriate privileges
+            _create_user(cursor, self.read_only_user, self._get_mysql_pass(self.read_only_user))
+            _grant_privileges(cursor, "*", self.read_only_user, "ALL")
+
+            # Create admin user with appropriate privileges
+            _create_user(cursor, self.admin_user, self._get_mysql_pass(self.admin_user))
+            _grant_privileges(cursor, "*", self.admin_user, "CREATE USER")
+
             self.mysql_provisioned = True
             self.save()
 
     def deprovision_mysql(self):
         """
-        Drop all MySQL databases.
+        Drop all MySQL databases and users.
         """
         if settings.INSTANCE_MYSQL_URL_OBJ and self.mysql_provisioned:
             cursor, connection = _get_mysql_cursor()
-            for database in self.mysql_database_names:
+
+            # Drop default databases and users
+            for database in self.mysql_databases:
                 # We can't use the database name in a parameterized query, the
-                # driver doesn't escape it properly. Se we escape it here instead
-                database_name = connection.escape_string(database).decode()
-                cursor.execute('DROP DATABASE IF EXISTS `{0}`'.format(database_name))
-            cursor.execute('DROP USER %s', (self.mysql_user,))
+                # driver doesn't escape it properly. So we escape it here instead
+                database_name = connection.escape_string(database["name"]).decode()
```

Comment:
Just like the `_create_database` method, I think it should be the `_drop_database` method itself that should do the escaping.